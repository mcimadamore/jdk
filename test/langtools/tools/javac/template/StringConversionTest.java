/*
 * @test /nodynamiccopyright/
 * @summary smoke test for string -> string template conversion
 * @enablePreview
 * @compile/fail/ref=StringConversionTest.out -XDrawDiagnostics StringConversionTest.java
 */
class StringConversionTest {
    static void testAssign(String prefix) {
        StringTemplate st1 = "Hello" + prefix; // bad
        StringTemplate st2 = "Hello"; // ok
        StringTemplate st3 = "{prefix}hello"; // ok
    }

    static void testCast(String prefix) {
        StringTemplate st1 = (StringTemplate) "Hello" + prefix; // bad
        StringTemplate st2 = (StringTemplate) "Hello"; // ok
        StringTemplate st3 = (StringTemplate) "{prefix}hello"; // ok
    }

    static void testCall(String prefix) {
        m("Hello" + prefix); // bad
        m("Hello"); // ok
        m("\{prefix}Hello"); // ok
    }

    static void testSwitch(String prefix) {
        switch ("Hello" + prefix) {
            case StringTemplate st -> { } // bad
        };
        switch ("Hello") {
            case StringTemplate st -> { } // ok
        };
        switch ("\{prefix}Hello") {
            case StringTemplate st -> { } // ok
        };
    }

    static void m(StringTemplate st) { }
}
