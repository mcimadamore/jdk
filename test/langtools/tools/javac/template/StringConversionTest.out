StringConversionTest.java:9:38: compiler.err.prob.found.req: (compiler.misc.inconvertible.string.template: java.lang.StringTemplate)
StringConversionTest.java:15:55: compiler.err.prob.found.req: (compiler.misc.inconvertible.string.template: java.lang.StringTemplate)
StringConversionTest.java:21:9: compiler.err.cant.apply.symbol: kindname.method, m, java.lang.StringTemplate, java.lang.String, kindname.class, StringConversionTest, (compiler.misc.no.conforming.assignment.exists: (compiler.misc.inconvertible.string.template: java.lang.StringTemplate))
StringConversionTest.java:28:18: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.String, java.lang.StringTemplate)
- compiler.note.preview.filename: StringConversionTest.java, DEFAULT
- compiler.note.preview.recompile
4 errors
